green_cli/authenticators/__init__.py

```diff
@@ -29,7 +29,8 @@ def login(self, session_obj):
         return gdk.login_user(session_obj, hw_device, json.dumps(credentials))
 
     def register(self, session):
-        return gdk.register_user(session, self.hw_device, self.mnemonic)
+        credentials = {'mnemonic': self.mnemonic} if self.mnemonic else {}
+        return gdk.register_user(session, self.hw_device, json.dumps(credentials))
```

Comment:
I think the `if hw_device == '{}':` logic from `login` should be put into a `get_credentials` function and then the result of calling that passed in both login and register cases.

I'm a little concerned about the password always being passed in credentials too, as with the corresponding gdk change it appears from code inspection that 'password' should only be present if there is mnemonic decryption to be done (even if the password is blank). Maybe this can be clarified with a test?

Reply:
> I think the if hw_device == '{}': logic from login should be put into a get_credentials function and then the result of calling that passed in both login and register cases

Done in a separate commit.

> password always being passed in credentials too

For now if the password is empty the mnemonic is not decrypted, https://github.com/Blockstream/gdk/blob/master/src/utils.cpp#L303-L307

Reply:
> For now if the password is empty the mnemonic is not decrypted.

Great, thanks for checking. I guess we should document that in gdk.